libd-llvm/src/d/llvm/type.d

```diff
@@ -302,7 +302,16 @@ final class TypeGen {
 	}
 	
 	LLVMTypeRef visit(Interface i) {
-		assert(0, "codegen for interface is not implemented.");
+		auto llvmStruct = LLVMStructCreateNamed(llvmCtx, i.mangle.toStringz());
+		auto structPtr = LLVMPointerType(llvmStruct, 0);
+		auto vtblStruct = LLVMStructCreateNamed(llvmCtx, (i.mangle ~ "__vtbl").toStringz());
+		auto vtblPtr = LLVMPointerType(vtblStruct, 0);
+		auto objectPtr = visit(pass.object.getObject());
+		LLVMTypeRef[2] elements;
+		elements[0] = objectPtr;
+		elements[1] = vtblPtr;
+		LLVMStructSetBody(structPtr, elements.ptr, elements.length, false);
+		return structPtr;
```

Comment:
Actually, I was returning a pointer because I was following the implementation for 'Class', which was returning a pointer. Guess it was wrong.